src/synapse/azext_synapse/custom.py

```diff
@@ -12,38 +13,59 @@
+                           main_class_name, executor_size, executors, language=SparkBatchLanguage.Scala,
+                           command_line_arguments=None,
+                           reference_files=None, archives=None, configuration=None,
                            tags=None, detailed=True):
-    dotnet_file = "local:///usr/hdp/current/spark2-client/jars/microsoft-spark.jar"
-    dotnet_class = "org.apache.spark.deploy.dotnet.DotnetRunner"
-
-    if language.upper() != "DOTNET" and (not file or not class_name):
-        raise CLIError('Scala and Python spark batch job must provide value for parameter file and class_name.'
-                       'If you want to create a DotNet spark batch job please add `--language DOTNET`.')
-
-    if language.upper() == "DOTNET":
-        file = dotnet_file
-        class_name = dotnet_class
+    file = main_definition_file
+    class_name = main_class_name
+    args = command_line_arguments
+    conf = configuration
+    if language.upper() == SparkBatchLanguage.SparkDotNet.value.upper() or \
+            language.upper() == SparkBatchLanguage.CSharp.value.upper():
+        file = DOTNET_FILE
+        class_name = DOTNET_CLASS
+
+        args = [main_definition_file, main_class_name]
+        if command_line_arguments:
+            args = args + command_line_arguments
+
+        archives = ["{}#{}".format(main_definition_file, SPARK_DOTNET_UDFS_FOLDER_NAME)] + archives if archives \
+            else ["{}#{}".format(main_definition_file, SPARK_DOTNET_UDFS_FOLDER_NAME)]
+
+        if not conf:
+            conf = {SPARK_DOTNET_ASSEMBLY_SEARCH_PATHS_KEY: './{}'.format(SPARK_DOTNET_UDFS_FOLDER_NAME)}
+        else:
+            conf[SPARK_DOTNET_ASSEMBLY_SEARCH_PATHS_KEY] = './{}'.format(SPARK_DOTNET_UDFS_FOLDER_NAME)
+
+    files = None
+    jars = None
+    if reference_files:
+        files, jars = categorized_files(reference_files)
+    driver_cores = EXECUTOR_SIZE[executor_size]['Cores']
+    driver_memory = EXECUTOR_SIZE[executor_size]['Memory']
+    executor_cores = EXECUTOR_SIZE[executor_size]['Cores']
+    executor_memory = EXECUTOR_SIZE[executor_size]['Memory']
 
     livy_batch_request = ExtendedLivyBatchRequest(
-        tags=tags,
-        name=job_name, file=file, class_name=class_name, args=args, jars=jars, files=files, archives=archives,
-        conf=conf, driver_memory=driver_memory, driver_cores=driver_cores, executor_memory=executor_memory,
-        executor_cores=executor_cores, num_executors=num_executors)
-
+        tags=tags, name=job_name, file=file, class_name=class_name, args=args, jars=jars,
+        files=files, archives=archives, conf=conf, driver_memory=driver_memory, driver_cores=driver_cores,
+        executor_memory=executor_memory, executor_cores=executor_cores, num_executors=executors)
+    print(str(livy_batch_request))
     return client.create(workspace_name, spark_pool_name, livy_batch_request, detailed)

```

Comment:
remove this line?

Reply:
so sorry. removed

---
In reply to: [417297374](https://github.com/Azure/azure-cli-extensions/pull/1581#discussion_r417297374) [](ancestors = 417297374)